test_runner/src/main/kotlin/ftl/run/platform/common/AfterRunTests.kt

```diff
@@ -11,37 +11,32 @@ import java.nio.file.Files
 import java.nio.file.Paths
 
 internal fun afterRunTests(
-    jobs: List<TestMatrix>,
+    testMatrices: List<TestMatrix>,
     runGcsPath: String,
     stopwatch: StopWatch,
     config: IArgs
-): MatrixMap {
-    val savedMatrices = mutableMapOf<String, SavedMatrix>()
-
-    jobs.forEach { matrix ->
-        val matrixId = matrix.testMatrixId
-        savedMatrices[matrixId] = SavedMatrix(matrix)
-    }
-
-    val matrixMap = MatrixMap(savedMatrices, runGcsPath)
+) = MatrixMap(
+    map = testMatrices.toSavedMatrixMap(),
+    runPath = runGcsPath
+).also { matrixMap ->
     updateMatrixFile(matrixMap, config)
     saveConfigFile(matrixMap, config)
 
-    println(FtlConstants.indent + "${savedMatrices.size} matrix ids created in ${stopwatch.check()}")
+    println(FtlConstants.indent + "${matrixMap.map.size} matrix ids created in ${stopwatch.check()}")
     val gcsBucket = "https://console.developers.google.com/storage/browser/" +
             config.resultsBucket + "/" + matrixMap.runPath
     println(FtlConstants.indent + gcsBucket)
     println()
-
-    return matrixMap
 }
 
+private fun List<TestMatrix>.toSavedMatrixMap() = this
+    .map { matrix -> matrix.testMatrixId to SavedMatrix(matrix) }
+    .toMap()
+
 private fun saveConfigFile(matrixMap: MatrixMap, args: IArgs) {
-    val configFilePath = if (args.useLocalResultDir()) {
-        Paths.get(args.localResultDir, FtlConstants.configFileName(args))
-    } else {
+    val configFilePath = if (args.useLocalResultDir())
+        Paths.get(args.localResultDir, FtlConstants.configFileName(args)) else
```

Comment:
TBH I don't like it when I need to read whole line to find out there is else at the end. But it is matter of personal preferences  :)

Reply:
TBH I don't like `if else` and prefer using `when` but our `detect` config forces to use `if else` in cases like this. Anyway I like that kind of formatting cause of clear indentation same for two parts.